sdk/include/opentelemetry/sdk/http/http_client.h

```diff
@@ -0,0 +1,139 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+#include "opentelemetry/http/http_client.h"
+#include "opentelemetry/version.h"
+
+#include <map>
+#include <vector>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace http
+{
+namespace http_api = opentelemetry::http;
+
+// HttpHeaders implementation
+class HttpHeaders : http_api::HttpHeaders, std::multimap<std::string, std::string>
+{
+
+public:
+  virtual void set(nostd::string_view const &name, nostd::string_view const &value) override
+  {
+    auto range = equal_range(std::string(name.data()));
+    auto hint  = erase(range.first, range.second);
+    insert(hint, std::make_pair(name, value));
+  }
+
+  virtual void add(nostd::string_view const &name, nostd::string_view const &value) override
+  {
+    insert(std::make_pair(name, value));
+  }
+
+  virtual nostd::string_view const &get(nostd::string_view const &name) const override
+  {
+    auto it = find(std::string(name.data()));
+    return (it != end()) ? it->second : m_empty;
+  }
+
+  virtual bool has(nostd::string_view const &name) const override
```

Comment:
I think `contains` is probably better name. Also it'd generally make sense to improve this to perform comparison ignoring case.. in most applications we'd better be ignoring case. HTTP headers are case-insensitive as per RFC2616.

Reply:
Changed code to use multimap with case insensitive comparison